File: Uni_App/app/Models/Announcement.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Announcement extends Model
{
    use HasFactory;

    protected $fillable = [
        'title',
        'content',
        'image_path',
        'target_audience',
        'target_college_id',
        'target_program_id',
        'is_active',
        'published_at',
        'expires_at',
    ];

    protected $casts = [
        'is_active' => 'boolean',
        'published_at' => 'datetime',
        'expires_at' => 'datetime',
    ];

    protected $appends = ['image_url'];

    public function getImageUrlAttribute()
    {
        return $this->image_path ? asset('storage/' . $this->image_path) : null;
    }

    /**
     * Get the college targeted by this announcement (if applicable).
     */
    public function targetCollege()
    {
        return $this->belongsTo(College::class, 'target_college_id');
    }

    /**
     * Get the program targeted by this announcement (if applicable).
     */
    public function targetProgram()
    {
        return $this->belongsTo(Program::class, 'target_program_id');
    }
}

File: Uni_App/student_portal/lib/features/announcements/models/announcement.dart
class Announcement {
  final int id;
  final String title;
  final String content;
  final String? imageUrl;
  final String targetAudience;
  final int? targetCollegeId;
  final int? targetProgramId;
  final bool isActive;
  final String? publishedAt;
  final String? expiresAt;
  final String? createdAt;
  final String? updatedAt;

  Announcement({
    required this.id,
    required this.title,
    required this.content,
    this.imageUrl,
    required this.targetAudience,
    this.targetCollegeId,
    this.targetProgramId,
    required this.isActive,
    this.publishedAt,
    this.expiresAt,
    this.createdAt,
    this.updatedAt,
  });

  factory Announcement.fromJson(Map<String, dynamic> json) {
    return Announcement(
      id: json['id'],
      title: json['title'],
      content: json['content'],
      imageUrl: json['image_url'],
      targetAudience: json['target_audience'],
      targetCollegeId: json['target_college_id'],
      targetProgramId: json['target_program_id'],
      isActive: json['is_active'] ?? true,
      publishedAt: json['published_at'],
      expiresAt: json['expires_at'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }
}
